### PR TITLE
LPS-181796 [TR2] Automation Test | master

### PR DESCRIPTION
**Pull request by joelrocha68**
Build Permissions
https://issues.liferay.com/browse/LPS-181795

### CODE REVIEW

**Comment by joelrocha68** [comment]
ci:test:sf

**Comment by joelrocha68** [comment]
ci:test:site-initializer-testray:nocompile

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+		}
+
+		task ("Then the build should not appear on the list") {
+			AssertElementNotPresent(
+				key_projectName = "Build Edited",
+				locator1 = "Testray#ITEM_NAME");
+		}
+	}
+
+	@description = "Story ID: LPS-178254 | Test Summary: Verify that the User with the 'Testray User' role should not be able to create/edit/delete the build but can view"
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase:259]: One test one intention. Please can you break this test down into 3?

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+		task ("And the user go to the Build page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+		}
```
Comment (luanakarla) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase:280]: Add to `Given` block.

**Review by luanakarla — CHANGES_REQUESTED** [review]
Hey @joelrocha68, thanks for working on it!

I requested some changes and suggested others. Can you take a look at this please?

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+			TestCase.setUpPortalInstance();
+
+			User.firstLoginPG();
+		}
+
+		task ("Create a new Testray Site") {
+			Testray.addTestraySite(siteName = ${testSiteName});
+		}
+
+		task ("Create all the necessary objects") {
```
[the remark this answers is not in the document]
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+		}
+
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@description = "Story ID: LPS-178254 | Test Summary: Verify that the User with the 'Testray Analyst' role should not be able to create/edit/delete the build but can view"
```
[the remark this answers is not in the document]
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+
+		task ("And the user go to the Build page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+		}
```
[the remark this answers is not in the document]
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+				key_kebabOption = "Promote",
+				locator1 = "ObjectAdmin#KEBAB_MENU_OPTION");
+
+			AssertElementNotPresent(
+				key_kebabOption = "Archive",
+				locator1 = "ObjectAdmin#KEBAB_MENU_OPTION");
+		}
+	}
+
+	@description = "Story ID: LPS-178254 | Test Summary: Verify that the User with the 'Testray Lead' role should be able to create/edit/delete the build"
```
[the remark this answers is not in the document]
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+		}
+
+		task ("And the user create a build") {
+			TestrayBuild.addBuildViaUI(
+				buildFieldName = "Build 2",
+				caseName = ${caseName},
+				dropdownItem = "New Build",
+				productVersionOption = "1.2",
+				routine = ${routineName});
+		}
```
[the remark this answers is not in the document]
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase (hunk 1 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase]: Done

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase (hunk 2 above)
Comment (joelrocha68) [reply modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase]: Done

**Comment by joelrocha68** [comment]
ci:test:sf

**Comment by joelrocha68** [comment]
ci:test:sf

**Review by luanakarla — APPROVED** [review]

**Comment by luanakarla** [comment]
LGTM!

**Comment by joelrocha68** [comment]
ci:forward

**Comment by joelrocha68** [comment]
ci:forward

**Comment by joelrocha68** [comment]
ci:forward

**Pull request closed without merging**